### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: koan/Koan/Parser.hs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+module Koan.Parser where
```
Comment (Kheldar) [inline koan/Koan/Parser.hs]: It would probably be worth giving some concrete example of something that could be parsed by this.

Comment (newhoggy) [inline koan/Koan/Parser.hs]: I think so too.

File: test/Check/Alternative.hs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import           Hedgehog
+import           Hedgehog.Extra
+import qualified Hedgehog.Gen     as Gen
+import qualified Hedgehog.Range   as Range
+
+{-# ANN module ("HLint: ignore Redundant do"  :: String) #-}
+
+prop_simple :: Property
+prop_simple = property $ do
+  True === True
```
Comment (Kheldar) [inline test/Check/Alternative.hs]: Tests?

File: test/Check/Parser.hs [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  K.runParser ka (fromString ea) === toKoanResult (P.parse pa (fromString ea))
+
+prop_peekChar :: Property
+prop_peekChar = property $ do
+  ea <- forAll $ Gen.string (Range.linear 0 10) (Gen.element "abcdef")
+  let ka = K.peekChar :: K.Parser (Maybe Char)
+  let pa = P.peekChar :: P.Parser (Maybe Char)
+  K.runParser ka (fromString ea) === toKoanResult (P.parse pa (fromString ea))
+
+-- prop_skip :: Property
```
Comment (Kheldar) [inline test/Check/Parser.hs]: Why is this stuff commented out?

Comment (newhoggy) [inline test/Check/Parser.hs]: Not written yet.

File: koan/Koan/Parser.hs (hunk 1 above)
Comment (Kheldar) [inline koan/Koan/Parser.hs]: I mean in the code - i.e. "Now that you've set up the machinery, try using it to parse `blah`".

**Pull request merged**